SingularityBase/src/main/java/com/hubspot/singularity/SingularityRequest.java

```diff
@@ -14,13 +14,14 @@
 import java.util.Set;
 
 @Schema(
-  description = "Settings that apply to all tasks and deploys assocaited with this request"
+  description = "Settings that apply to all tasks and deploys associated with this request"
 )
 public class SingularityRequest {
   private final String id;
   private final RequestType requestType;
   private final Optional<List<String>> owners;
   private final Optional<Integer> numRetriesOnFailure;
+  private final Optional<Integer> maxScale;
```

Comment:
I'm not entirely clear on how this new option would interact with autoscale

Reply:
it looks like AutoScale uses SingularityClient's `scaleSingularityRequest` ([code](https://git.hubteam.com/HubSpot/AutoScale/blob/master/AutoScaleData/src/main/java/com/hubspot/autoscale/data/externalservice/SingularityService.java#L410)), which hits this [resource](https://github.com/HubSpot/Singularity/blob/master/SingularityService/src/main/java/com/hubspot/singularity/resources/RequestResource.java#L1660)

Reply:
That resource uses SingularityValidator's `checkScale`, so you should the `checkBadRequest` also to that method so that AutoScale is blocked

(or make a `checkBadScaleRequest` method for both validator methods to use)

Reply:
It looks like [`checkScale`](https://github.com/HubSpot/Singularity/blob/76c383f587c22cd1ce21b2c837046326c8b3b2d5/SingularityService/src/main/java/com/hubspot/singularity/data/SingularityValidator.java#L1183) doesn't currently check against the global max scale in `MesosConfig`, so AutoScale isn't blocked by that when scaling. Do we want it be blocked by the new request-level max scale option?

Reply:
Good catch — we should check both max scale values. Since Singularity has been missing this, we should verify that no active requests would be caught by the current `maxNumInstancesPerRequest` value before this is rolled out.